crates/lovely-core/src/patch/regex.rs

```diff
@@ -43,20 +49,63 @@ impl RegexPatch {
         }
 
         let input = Input::new(rope.into_cursor());
-        let re = Regex::new(&self.pattern)
+        let re = Regex::builder()
+            .syntax(
+                 syntax::Config::new()
+                    .multi_line(true)
+                    .crlf(true)
+                    .ignore_whitespace(self.verbose)
+            )
+            .build(&self.pattern)
             .unwrap_or_else(|e| panic!("Failed to compile Regex '{}': {e:?}", self.pattern));
 
         let mut captures = re.captures_iter(input).collect_vec();
         if captures.is_empty() {
-            log::warn!("Regex '{}' on target '{target}' resulted in no matches", self.pattern);
+            log::warn!("Regex '{}' on target '{target}' resulted in no matches", self.pattern.escape_debug());
             return false;
         }
         if let Some(times) = self.times {
             if captures.len() < times {
-                log::warn!("Regex '{}' on target '{target}' resulted in {} matches, wanted {}", self.pattern, captures.len(), times);
+                if self.pattern.lines().count() > 1 {
+                    for line in format!(
+                        "Regex '''\n{}''' on target '{target}' resulted in {} matches, wanted {}",
+                        self.pattern,
+                        captures.len(),
+                        times
+                    )
+                    .lines()
+                    {
+                        log::warn!("{}", line);
+                    }
+                } else {
+                    log::warn!(
+                        "Regex '{}' on target '{target}' resulted in {} matches, wanted {}",
+                        self.pattern,
+                        captures.len(),
+                        times
+                    );
+                }
             }
             if captures.len() > times {
-                log::warn!("Regex '{}' on target '{target}' resulted in {} matches, wanted {}", self.pattern, captures.len(), times);
+                if self.pattern.lines().count() > 1 {
+                    for line in format!(
+                        "Regex '''\n{}''' on target '{target}' resulted in {} matches, wanted {}",
+                        self.pattern,
+                        captures.len(),
+                        times
+                    )
+                    .lines()
+                    {
+                        log::warn!("{}", line);
+                    }
+                } else {
+                    log::warn!(
+                        "Regex '{}' on target '{target}' resulted in {} matches, wanted {}",
+                        self.pattern,
+                        captures.len(),
+                        times
+                    );
+                }
```

Comment:
Code duplication - a couple times would be fine but 4 is too much.

Reply:
Factored out into its own (internal) function.